src/Hcal/HcalGeometryProvider.cxx

```diff
@@ -0,0 +1,132 @@
+#include "DetDescr/HcalGeometry.h"
+#include "Framework/ConditionsObjectProvider.h"
+#include "Framework/RunHeader.h"
+#include "Framework/EventHeader.h"
+#include <sys/types.h>
+#include <regex.h>
+
+/**
+ * @file HcalGeometryProvider.cxx
+ * @brief Class that creates HcalGeometry object based on python specification
+ * @author Jeremiah Mans, UMN
+ * @author Cristina Suarez, Fermi National Accelerator Laboratory
+ */
+
+namespace hcal {
+
+class HcalGeometryProvider : public framework::ConditionsObjectProvider {
+ public:
+  /**
+   * Class constructor
+   * @param parameters -- uses the "HcalGeometry" section to configure the HcalGeometry
+   */	
+  HcalGeometryProvider(const std::string& name,
+		       const std::string& tagname,
+		       const framework::config::Parameters& parameters,
+		       framework::Process& process);
+
+
+  /** Destructor */
+  virtual ~HcalGeometryProvider();
```

Comment:
Use
```cpp
virtual ~HcalGeometryProvider() = default;
```
instead. Safer and modern.

Reply:
Reminding you of this comment.

Reply:
My mistake. This shouldn't be defined as a deafult; it is defined below.

Reply:
Shouldn't be virtual.